---
title: "RNA-seq, Ribo-seq and proteomic "
date: "`r format(Sys.time(), 'Last Modified: %d-%b-%Y %R %p')`"
subtitle: "How does gene expression vary at different expression scales?"
author: "[Elie Teyssonniere](https://github.com/elielink/)"
mail: "elieteyssonniere@gmail.com"
github: "elielink"
home: "https://github.com/elielink/"
output:
  rmdformats::downcute:
    self_contained: true
    thumbnails: false
    lightbox: true
    gallery: true
    highlight: tango
    code_download: true
    code_folding: hide
    fig_caption: yes
    keep_md: yes
    toc_depth: 3
    toc_float:
      toc_collapsed: true
    number_sections: true
    theme: lumen
  editor_options: 
    chunk_output_type: console
---

```{r setup, include=FALSE, echo=F, warnings=F, message=F}
knitr::opts_chunk$set(echo = T, collapse = T, cache.lazy = F,autodep = T)
suppressPackageStartupMessages({
  library(xfun)
  # Load  package ---------------------------------------
  dep.pkg = c("corrplot",
"ggtree",
"gtools",
"pbapply",
"gridExtra",
"GGally",
"dplyr",
"performance", 
"ggplot2",
"ggfortify",
"tidyverse",
"gplots",
"data.table",
"ggpubr",
"ggcorrplot",
"plotly",
"ggsignif",
'ape')

  xfun::pkg_attach2(dep.pkg)

  source(here("src","functions_transcriptomics.R"))

})
# turn off annoying messages from dplyr::summarise
options(dplyr.summarise.inform = FALSE)
``` 

# General dataset summary {.tabset .tabset-pills}


## Our datasets

**_RNA-seq & Ribo-seq of 8 natural isolates of Saccharomyces cerevisiae_**

We performed both ribosome profiling and RNA sequencing on eight _S. cerevisiae_ natural isolates coming from very diverse ecological environments and being genetically strongly different ([Peter et al., 2018](https://www.nature.com/articles/s41586-018-0030-5)). These isolates were cultivated in Synthetic Complete (SC) up to mid-log phase, harvested and flash-frozen. The RNA-seq and Ribo-seq experiment was performed in collaboration with the Riken institute in Japan.
```{r load-rna-ribo-data}
## Our dataset
isolates = c('CPI','CMP','AMH','CQC','BPL','BTT','BED','BAN')
RNA_seq_data = readRDS(here("data",'RNA_seq.RDS'))
Ribo_seq_data = readRDS(here("data",'Ribo_seq.RDS'))
n = nrow(RNA_seq_data)
strains = fread('strains.csv',data.table = F)
strains = strains[strains$`Standardized name`%in%isolates,c(1,2,3,4,9,10,11,16)]
library(DT)
datatable(strains, rownames = FALSE, caption = NULL,
               filter = "top", escape = FALSE, style = "default",
               width = NULL, height = NULL)
```

Our data encompassed **`r n`** genes. The data was normalized using TPM normalization where for each gene in each isolate, we divided the raw count by the ORF length and we applied a per million factor (total read count / 1,000,000)


```{r corr-rna-ribo}

p1= ggcorr(RNA_seq_data ,method = c("everything", "spearman"), label = T, label_round = 2, midpoint = 0.75, limits = c(0.6,1), size = 2)+
  ggtitle('RNA-seq correlation matrix')
p2=ggcorr(Ribo_seq_data ,method = c("everything", "spearman"), label = T, label_round = 2, midpoint = 0.75, limits = c(0.6,1), size = 2)+
  ggtitle('Ribo-seq correlation matrix')

grid.arrange(p1,p2, ncol=2)
```

We also calculated a Translation efficiency value which correspond, for each gene in each isolate, to the Ribo-seq TPM value divided by the RNA-seq TPM value. In brief, this represents how well a transcript will be used for translation.

## Overlap with proteomic

```{r overlap-proteomics}

```
