---
title: "RNA-seq, Ribo-seq and proteomic "
date: "`r format(Sys.time(), 'Last Modified: %d-%b-%Y %R %p')`"
subtitle: "How does gene expression vary at different expression scales?"
author: "[Elie Teyssonniere](https://github.com/elielink/)"
mail: "elieteyssonniere@gmail.com"
github: "elielink"
home: "https://github.com/elielink/"
output:
  rmdformats::downcute:
    self_contained: true
    thumbnails: false
    lightbox: true
    gallery: true
    highlight: tango
    code_download: true
    code_folding: hide
    fig_caption: yes
    keep_md: yes
    toc_depth: 3
    toc_float:
      toc_collapsed: true
    number_sections: true
    theme: lumen
  editor_options: 
    chunk_output_type: console
---

<!-- Github Logo -->
<a href="https://github.com/elielink/" class="github-corner" aria-label="Visit my Github">
  <svg width="80" height="80" viewBox="0 0 250 250" style="fill:#69b3a2; color:#fff; position: absolute; top: 0; border: 0; right: 0;" aria-hidden="true">
    <path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path>
    <path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2" fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path>
    <path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z" fill="currentColor" class="octo-body"></path>
  </svg>
</a>

<style> 
.github-corner:hover .octo-arm{animation:octocat-wave 560ms ease-in-out}
@keyframes octocat-wave{0%,100%{transform:rotate(0)}20%,60%{transform:rotate(-25deg)}40%,80%{transform:rotate(10deg)}}
@media (max-width:500px){.github-corner:hover .octo-arm{animation:none}.github-corner .octo-arm{animation:octocat-wave 560ms ease-in-out}}
</style>

```{r setup, include=FALSE, echo=F, warnings=F, message=F}
knitr::opts_chunk$set(echo = T, collapse = T, cache.lazy = F,autodep = T)
suppressPackageStartupMessages({
  library(xfun)
  # Load  package ---------------------------------------
  dep.pkg = c("corrplot",
"ggtree",
"gtools",
"pbapply",
"gridExtra",
"GGally",
"dplyr",
"performance", 
"ggplot2",
"ggfortify",
"tidyverse",
"gplots",
"data.table",
"ggpubr",
"ggcorrplot",
"plotly",
"ggsignif",
'ape')

  xfun::pkg_attach2(dep.pkg)

  source("functions.R")

})
# turn off annoying messages from dplyr::summarise
options(dplyr.summarise.inform = FALSE)
``` 

# General dataset summary {.tabset .tabset-pills}


## Our datasets

**_RNA-seq & Ribo-seq of 8 natural isolates of Saccharomyces cerevisiae_**

We performed both ribosome profiling and RNA sequencing on eight _S. cerevisiae_ natural isolates coming from very diverse ecological environments and being genetically strongly different ([Peter et al., 2018](https://www.nature.com/articles/s41586-018-0030-5)). These isolates were cultivated in Synthetic Complete (SC) up to mid-log phase, harvested and flash-frozen. The RNA-seq and Ribo-seq experiment was performed in collaboration with the Riken institute in Japan.
```{r dataset}
# General dataset summary {.tabset .tabset-pills}

## Our dataset

isolates = c('CPI','CMP','AMH','CQC','BPL','BTT','BED','BAN')
RNA_seq_data = readRDS('RNA_seq.RDS')
Ribo_seq_data = readRDS('Ribo_seq.RDS')
n = nrow(RNA_seq_data)
strains = fread('strains.csv',data.table = F)
strains = strains[strains$`Standardized name`%in%isolates,c(1,2,3,4,9,10,11,16)]
require(DT)
datatable(strains, rownames = FALSE, caption = NULL,
               filter = "top", escape = FALSE, style = "default",
               width = NULL, height = NULL)
```

Our data encompassed **`r n`** genes. The data was normalized using TPM normalization where for each gene in each isolate, we divided the raw count by the ORF length and we applied a per million factor (total read count / 1,000,000)


```{r dataset2}
# General dataset summary {.tabset .tabset-pills}

## Our dataset


p1= ggcorr(RNA_seq_data ,method = c("everything", "spearman"), label = T, label_round = 2, midpoint = 0.75, limits = c(0.6,1), size = 2)+
  ggtitle('RNA-seq correlation matrix')
p2=ggcorr(Ribo_seq_data ,method = c("everything", "spearman"), label = T, label_round = 2, midpoint = 0.75, limits = c(0.6,1), size = 2)+
  ggtitle('Ribo-seq correlation matrix')

grid.arrange(p1,p2, ncol=2)
```

We also calculated a Translation efficiency value which correspond, for each gene in each isolate, to the Ribo-seq TPM value divided by the RNA-seq TPM value. In brief, this represents how well a transcript will be used for translation.

## Overlap with proteomic

```{r dataset3}
# General dataset summary {.tabset .tabset-pills}

## Overlap with proteomic



```

# Session Info {-}

```{r session-info}
sessionInfo()
```