---
output: html_document
editor_options: 
  chunk_output_type: console
---
```{r load-setup-data, echo=F}
chap_cur=01
chap_num = sprintf("%02d-",chap_cur)
load(here::here('output',sprintf("%02d-yeast-data.rdata",chap_cur-1)))
```

# Transcriptomics {.tabset .tabset-pills}

## Summary

During evolution, mutations arise in genomes, potentially affecting every gene
and protein at every site over time. Degrees of sequence divergence within
populations and between species reflect the interplay between several
interconnected constraints (e.g., genomic, transcriptional, functional,
biophysical, structural, environmental…) and underlie phenotypic diversity and
organism fitness. Protein expression is the main determinant of a protein’s
sequence conservation (e.g., the more a gene is expressed, the more its sequence
is conserved). Although this expression-conservation correlation is well
established, its mechanistic origin is not well understood.

In this perspective, analyzing variation of gene expression across every layer
could reveal the key processes driving evolution among several yeast (*S. cerevisiae*) 
isolates sharing an almost identical genetic background, except for
newly acquired mutations, notably in protein-coding genes (e.g. Single
Nucleotide Variations).

Gene expression is a multi-layered phenomenon in which variations occur 
differently at each step of the protein synthesis and life
(transcription, translation and protein abundance). The exploration of each
of these steps is therefore essential to define the constraints affecting the
final protein abundance. Given this, we sought to understand how individuals
differ in terms of translational regulation and if the translational variation
is related to the transcriptional variation. 

Therefore, we performed a joint exploration of RNA-seq, ribosome and proteomics
profiling on 8 natural *Saccharomyces cerevisiae* yeast isolates with 
widely diverse genetic and ecologic origins. We found that the evolutionary
constraints on gene expression showed different magnitudes depending on the
expression layer. Specifically, expression variations tended to be 10% lower at
the translational level, which was related to a phenomenon called
post-transcriptional buffering. This phenomenon was more likely to affect some
specific genes such as essential genes as well as genes related to protein
complexes. Surprisingly, genes being preferentially affected by the
post-transcriptional buffering tended to be less expressed than the genes not
affected by this phenomenon. Together, these results highlight that variations
in gene expression were shaped differently depending on the expression level and
multiple factors, including certain functional and physical constraints. We have
also been collecting proteomics data for these 8 isolates, which will allow us
to assess with high confidence the balance between biophysics (in particular
abundance) versus function on protein evolution.


## Our datasets

**_RNA-seq & Ribo-seq of 8 natural isolates of Saccharomyces cerevisiae_**

We performed both ribosome profiling and RNA sequencing on eight _S. cerevisiae_
natural isolates coming from very diverse ecological environments and being
genetically strongly different ([Peter et al.,2018](https://www.nature.com/articles/s41586-018-0030-5)).
These isolates were cultivated in Synthetic Complete (SC) up to mid-log phase, 
harvested and flash-frozen. The RNA-seq and Ribo-seq experiment was performed in
collaboration with the Riken institute in Japan.

```{r load-rna-ribo-data}
## Our dataset
isolates = c('CPI','CMP','AMH','CQC','BPL','BTT','BED','BAN') %>% sort
RNA_seq_data = readRDS(here("data",'RNA_seq.RDS')) %>% relocate(all_of(isolates))
Ribo_seq_data = readRDS(here("data",'Ribo_seq.RDS')) %>% relocate(all_of(isolates))
n = nrow(RNA_seq_data)
strains = fread(here("data",'strains.csv'),data.table = F)
strains = strains[strains$`Standardized name`%in%isolates,c(1,2,3,4,9,10,11,16)]
library(DT)
datatable(strains, rownames = FALSE, caption = NULL,
               filter = "top", escape = FALSE, style = "default",
               width = NULL, height = NULL)
```

Our data encompassed **`r n`** genes. The data was normalized using TPM
normalization where for each gene in each isolate, we divided the raw count by
the ORF length and we applied a per million factor (total read count /
1,000,000)


```{r corr-rna-ribo}

p1= ggcorr(RNA_seq_data ,method = c("everything", "spearman"), label = T, label_round = 2, midpoint = 0.75, limits = c(0.6,1), size = 2)+
  ggtitle('RNA-seq correlation matrix')
p2=ggcorr(Ribo_seq_data ,method = c("everything", "spearman"), label = T, label_round = 2, midpoint = 0.75, limits = c(0.6,1), size = 2)+
  ggtitle('Ribo-seq correlation matrix')

grid.arrange(p1,p2, ncol=2)
```

We also calculated a Translation efficiency value which correspond, for each
gene in each isolate, to the Ribo-seq TPM value divided by the RNA-seq TPM
value. In brief, this represents how well a transcript will be used for
translation.

## Overlap with proteomic

```{r overlap-proteomics}

proteomic_WIS2 = fread(here::here('output','median-normalized.txt'), data.table = F)
n = nrow(proteomic_WIS2)
a = fread(here::here('output','proteomics-normalized-log10_intensities.tsv'))
rownames(proteomic_WIS2)= a$UNIPROT

prot_gene_name = fread(here::here('output','test.csv'), fill = T, data.table = F) %>% 
                  mutate(Chromosome = recode(Chromosome,'(3)'=V8) ) %>% 
                  dplyr::select(-V8) %>%
                  dplyr::filter(!duplicated(prot_names))
                  

# Calculate mean of normalized intensity per strain
int_strains = pivot_longer(proteomic_WIS2 %>% rownames_to_column('uniprot'), 
             -uniprot, values_to='int2use',
                       names_to = c('strain','bio','tech','day'),
                       names_pattern = "([^_]+)_([^_]+)_([^_]+)_([^_]+)") %>% 
      group_by(uniprot,strain=toupper(strain)) %>% summarize( mean_int = mean_(int2use)) %>%
      pivot_wider(id_cols=uniprot, names_from = strain, names_glue = "{strain}", values_from = mean_int )

proteomic_WIS_mean_filtered = data.frame(int_strains) %>% drop_na %>% 
                      dplyr::filter(uniprot %in% prot_gene_name$prot_names) %>% 
                      left_join(prot_gene_name[,c('prot_names','ID')], by=c('uniprot'='prot_names'))

all_exp = proteomic_WIS_mean_filtered %>% 
  inner_join(RNA_seq_data %>% rownames_to_column('ID'), by ='ID', suffix=c('_Prot','')) %>% 
  inner_join(Ribo_seq_data %>% rownames_to_column('ID'), by ='ID', suffix=c('_RNA','_Ribo'))

log2_all_exp= all_exp %>% column_to_rownames('ID') %>%
              # transform to log2 to RNA and Ribo (already done for normalized proteomics)
              mutate( across( ends_with(c('_RNA','_Ribo')), .fns = log2) ) %>%
              dplyr::select(where(is.numeric))
id_exp = rownames(log2_all_exp)

boxplot(log2_all_exp,las=2,ylab='Gene expression (log2)')
```

Our proteomic data encompassed **`r n`** genes. The overlap between the two datasets encompassed **`r nrow(log2_all_exp)`** (after several filtration of some missing values). The data between the different data sets needed to be normalized so it can be studied together (see graph above). We therefore performed quantile normalization to obtain equally distributed data (see below)

```{r quantile-norm}
index_to_mean <- function(my_index, my_mean){ return(my_mean[my_index]) }

df_rank <- apply(log2_all_exp,2,rank,ties.method="min") 
df_sorted <- data.frame(apply(log2_all_exp, 2, sort))
df_mean <- apply(df_sorted, 1, mean)

log2_all_exp_norm <- apply(df_rank, 2, index_to_mean, my_mean=df_mean) %>% as_tibble
rownames(log2_all_exp_norm) = id_exp 
boxplot(log2_all_exp_norm,las=2, 
        ylab='Normalized gene expression (log2)', 
        title='Quantile-normalized gene expression across RNA/Ribo/Proteomics')

prot_norm = dplyr::select(log2_all_exp_norm, ends_with('_Prot')) %>% 
            rename_with(.fn=str_remove_all,pattern = "_Prot") 
rna_norm = dplyr::select(log2_all_exp_norm, ends_with('_RNA')) %>% 
          rename_with(.fn=str_remove_all,pattern = "_RNA") 
ribo_norm = dplyr::select(log2_all_exp_norm, ends_with('_Ribo')) %>% 
            rename_with(.fn=str_remove_all,pattern = "_Ribo") 

```

# Post transcriptional buffering {.tabset .tabset-pills}

The post-transcriptional buffering (PTB) is a phenomenon where transcriptional variations tend to be buffered as the expression process progresses (add ref). It has been observed in different situations... To be completed

It is possible to detect this phenomenon in different ways:

## Correlation between isolate

Using the quantile normalized data, we performed Spearman correlation between all the isolate profile in each expression level separately (using the median value of the replicate for the proteomic data)

Comparing the correlation values, it reveals that the profiles tended to be more and more similar as the expression process progresses (increasing of the correlation coefficient values), bringing a first proof of the presence of the phenomenon across our 8 isolates. 

```{r PTB1}
library(corrr)
cor_prot= prot_norm %>%  correlate(method = 'sp') %>% column_to_rownames('term')
cor_rna= rna_norm %>%  correlate(method = 'sp') %>% column_to_rownames('term')
cor_ribo= ribo_norm %>%  correlate(method = 'sp') %>% column_to_rownames('term')
get_lower_tri = function(m){ m[lower.tri(m)] }

cor_exp= tibble(RNA = get_lower_tri(cor_rna), Ribo = get_lower_tri(cor_ribo), Prot = get_lower_tri(cor_prot))
df_cor_exp = pivot_longer(cor_exp, everything()) %>% 
              arrange(name) %>% mutate(experiment = factor(name))
pair_exp = unique(df_cor_exp$experiment) %>% as.character() %>% combn(m = 2) %>% as_tibble()

ggplot(df_cor_exp,aes(x=experiment, y=value,fill=experiment))+
  geom_boxplot()+
  theme_classic()+
  xlab('')+
  ylab('Rho (spearman)')+
  stat_compare_means(comparisons =pair_exp)
```


## Variation quantification 

It is also possible to explore PTB by quantifying the variation in each pairwise comparison. Basically, for each gene in each isolate pairwise comparison, we use the absolute value of the log2 transformed fold to represent the intensity of the expression variation between the 2 isolates. The more this value increases, the more a gene displays variable regulation between isolate.

We calculated this value in each pairwise comparison, for each gene and at each expression level. We found that the intensities of the variations were decreasing as long as the expression process progresses, supporting once again the presence of the PTB phenomenon in our dataset.

```{r PTB2}
#remotes::install_github("TimTeaFan/dplyover")
library(dplyover)

# calculate fold change pairwise for each experiment
fc_prot = transmute(prot_norm, across2x(everything(), everything(),.fns = ~.x/.y,.comb = 'minimal'),
            experiment='Prot')
fc_rna = transmute(rna_norm, across2x(everything(), everything(),.fns = ~.x/.y,.comb = 'minimal'),
            experiment='RNA')
fc_ribo = transmute(ribo_norm, across2x(everything(), everything(),.fns = ~.x/.y,.comb = 'minimal'),
            experiment='Ribo') 
# combine all pairwise foldchange
fc_exp = bind_rows(fc_prot,fc_rna,fc_ribo) %>% 
          pivot_longer(-experiment,names_to='pair',values_to = 'fc') %>%
          arrange(experiment,pair) %>% 
          mutate(log2fc = abs(log2(fc)))

pair_exp=unique(fc_exp$experiment) %>% as.character() %>% combn(m = 2) %>% as_tibble()

ggplot(fc_exp,aes(experiment, log2fc, fill=experiment))+
  geom_boxplot()+
  scale_y_log10()+
  stat_compare_means(comparisons = pair_exp)+
  theme_classic()+
  xlab('')+
  ylab('|log2(FC)|')

```

## Euclidean distances between the profiles

Euclidean distances can also give information on how variable datasets can be. We once again used the quantile normalized data to calculate Euclidean distances between the profile in each expression level.

Consistently to the previous results, we found that the distances were lower and lower as the expression process progresses, suggesting once again that the expression variations were decreased. This also supported the presence of the PTB phenomenon

```{r PTB3}
# Calculate euclidean distance between normalized strains expression
dist_exp = tibble( 
  'Prot'= prot_norm %>% t() %>% dist %>% as.matrix %>% get_lower_tri(),
  'RNA'= rna_norm %>% t() %>% dist %>% as.matrix %>% get_lower_tri(),
  'Ribo'= ribo_norm %>% t() %>% dist %>% as.matrix %>% get_lower_tri()
)
# convert ot long format
df_dist_exp = pivot_longer(dist_exp, everything(),names_to='experiment',values_to = 'euclidean') %>%
              arrange(experiment)

pair_exp=unique(df_dist_exp$experiment) %>% as.character() %>% combn(m = 2) %>% as_tibble()

ggplot(df_dist_exp,aes(x=experiment,y=euclidean, fill=experiment))+
  geom_boxplot()+
  theme_classic()+
  ylab('Euclidean distance')+
  xlab('')+
  stat_compare_means(comparisons = pair_exp)

```

## Variance comparison

A simple variance calculation can also give a clue on the variation intensity of gene expression. We calculated the variance in all dataset for each gene and found that it tended to be higher at the transcription level, lower at the protein abundance level (with the translation level being in the middle), which was in accordance with the previous exploration

```{r PTB4}
var_exp = tibble( id = id_exp,
                  Prot =  apply(prot_norm,1,var),
                  RNA = apply(rna_norm,1,var),
                  Ribo =apply(ribo_norm,1,var)) %>% 
          pivot_longer(-id, names_to='experiment',values_to='variance') %>%
          arrange(experiment,id)

pair_exp=unique(var_exp$experiment) %>% as.character() %>% combn(m = 2) %>% as_tibble()

ggplot(var_exp,aes(x=experiment,y=variance, fill=experiment))+
  geom_boxplot()+
  theme_classic()+
  ylab('Variance')+
  xlab('')+
  stat_compare_means(comparisons = pair_exp)+
  scale_y_log10()
```

# Expression evolution on the different layers {.tabset .tabset-pills}

Expression evolution can be explored in each layer using several methods. One of them uses the construction of phylogeny trees using the expression data at each expression layer (transcriptome, translatome, proteome). This method has been used previously for expression evolution exploration in mammals ([Wang et al., 2020](https://www.nature.com/articles/s41586-020-2899-z)). We tried to applied this method to our data sets. We used the quantile normalized data (encompassing **`r nrow(log2_all_exp_norm)`** genes).

## Transcriptome, translatome and proteome trees

```{r tree1}
par(mfrow= c(1,3))

tree_exp = tibble( 
  'Prot'= prot_norm %>% t() %>% dist %>% nj %>% .[['edge.length']] %>% sum,
  'RNA'= rna_norm %>% t() %>% dist %>% nj %>% .[['edge.length']] %>% sum,
  'Ribo'= ribo_norm %>%  t() %>% dist %>% nj %>% .[['edge.length']] %>% sum
)

#test = modelr::bootstrap(log2_all_exp_norm,1000)

a =lapply(c('Prot', 'RNA','Ribo'), function(i){
  i<<-i
b = dist(t(log2_all_exp_norm[, grep(i, colnames(log2_all_exp_norm))]))
a = nj(b)
c = sum(a$edge.length)

d =lapply(1:1000, function(j){
  b = dist(t(log2_all_exp_norm[sample(x=1:nrow(log2_all_exp_norm),size = nrow(log2_all_exp_norm),replace = T), grep(i, colnames(log2_all_exp_norm))]))
  a = nj(b)
  
  sum(a$edge.length)
})
d= unlist(d)
c=c(c,qnorm(0.975)*sd(d)/sqrt(length(d)))
c= setNames(c,c('length', 'err'))

plot(a,type='u', main = i)
c
})
a= do.call(rbind,a)
rownames(a)= c('Prot', 'RNA','Ribo')
a= as.data.frame(a)
a$level=rownames(a)
a$level=factor(a$level,levels = c('RNA','Ribo','Prot'))
```

The measure of the global expression evolution can be achieved using the total length of the tree (see below, error bar correspond the to 95% confidence interval of the tree length using 1000 steps boot-straping)

```{r tree2}

ggplot(a, aes(level, fill=level, y= length))+
  geom_bar(stat = 'identity',width = 0.5, alpha= 0.8)+
  geom_errorbar(aes(x=level, ymin=length-err, ymax=length+err), width=0.1)+
  theme_classic()

```

Consistently with the detection of the post-transcriptional buffering, the expression evolution seems to be more and more constrained at each step of gene expression (Constraint RNA < Constraint Ribo < Constraint Prot).

## Characteristic influencing gene expression evolution

It is possible to explore how some gene characteristics can influence gene expression evolution by comparing the length of a tree generated using the gene having these characteristics with tree lengths obtain with all or random set of genes. However, it necessary to first check if the number of genes used to generate a tree influence the final tree length. By sub sampling the number of gene to generate the trees (100 iterations with each possible number of gene starting from 10) it is possible to detect an effect on final tree length. It is therefore important to take into account the number of gene tested and compare this to trees generated with the same amount of genes. 

```{r tree3}

#d=lapply(rep(10:nrow(df_final2),100),function(j){
 # 
#  temp= df_final2[sample(1:nrow(df_final2),j),]
#  a =lapply(c('Prot', 'RNA','Ribo'), function(i){
#    i<<-i
#    b = dist(t(temp[, grep(i, colnames(temp))]))
#    a = nj(b)
#    c = sum(a$edge.length)
#    
#  })
#  a= t(data.frame(length=a))
 # rownames(a)= c('Prot', 'RNA','Ribo')
#  a= as.data.frame(a)
  #a$num=j
#  return(a)
#})
#d= do.call(rbind,d)
#d$type= NA
#d[grep('Prot', rownames(d)),"type"]='Prot'
#d[grep('RNA', rownames(d)),"type"]='RNA'
#d[grep('Ribo', rownames(d)),"type"]='Ribo'
#d$type=factor(d$type,levels = c('RNA','Ribo','Prot'))
#write_rds(d,'control_tree_length.RDS')
ctrl_tree_len= read_rds(here('data','control_tree_length.RDS'))
ggplot(ctrl_tree_len,aes(x=num,y=V1, col= type))+
  geom_point(alpha=0.6)+
  theme_classic()+
  xlab('Number of genes')+
  ylab("Tree length")


control_tree_length = ctrl_tree_len
a=gene_feature[,(sapply(gene_feature, class)=='logical')]
a =as.data.frame(a)
rownames(a)=gene_feature$ORF
a = a[base::intersect(rownames(a), rownames(log2_all_exp_norm)),]
a=a[,sapply(a,sum)>9]
```



We'll be using gene characteristics (quantified with Boolean values: TRUE or FALSE) generated with several databases (see [Yeastomic](https://github.com/benjamin-elusers/yeastomics)). Not all genes were encompassed and therefore we'll be using the gene overlapping between the expression datasets and the characteristic database (**`r nrow(a)`** genes). To detect whether or not a characteristic influences the gene expression evolution, we compared the length L of a tree obtained with N genes displaying the characteristic with lengths obtained with 100 trees generated by randomly select N genes. The 100 length being globally normally distributed, it is possible to obtain the Cumulative Distribution Function the length L compared to the lengths of the 100  random trees. For instance, the characteristic _cat_genomics.sgd.chr_A_ (which is: is the gene on the chromosome 1?) encompassed **`r sum(a[,1])`** gene in the our dataset (**`r sum(a[,1])`** out of **`r nrow(a)`** genes overlapping between our expression data and the characteristic database). 
```{r tree4}

b = nj(dist(t(log2_all_exp_norm[rownames(log2_all_exp_norm)%in%rownames(a)[a$cat_genomics.sgd.chr_A], grep('RNA', colnames(log2_all_exp_norm))])))
b=sum(b$edge.length)
c= pnorm(b,mean = mean(control_tree_length[control_tree_length$num==sum(a[,1])&control_tree_length$type=='RNA',"V1"]),sd =sd(control_tree_length[control_tree_length$num==sum(a[,1])&control_tree_length$type=='RNA',"V1"]) )
```
The length of the tree generated with these 30 gene using RNA-seq data is **`r b`**. The probability (not corrected) of getting this length with random trees = **`r c`**

```{r tree5}
par(mfrow=c(1,1))
plot(density(control_tree_length[control_tree_length$num==sum(a[,1])&control_tree_length$type=='RNA',"V1"]),main = 'Tree length distribution')
abline(v=b, col='red')
```

We performed this analysis for **`r ncol(a)`** characteristics at each expression level. We corrected the probability values (false discovery rate correction) and using a corrected probability value threshold of 0.01 to select the characteristic impacting gene expression.

```{r tree6}
b= apply(a,2, function(i){
  i<<-i
  b = rownames(a)[i]
  c = log2_all_exp_norm[b,]
  d =sapply(c('Prot', 'RNA','Ribo'), function(j){
    j<<-j
    b = dist(t(c[, grep(j, colnames(c))]))
    a = nj(b)
    e = sum(a$edge.length)
    
    return(e)
  })
  e =control_tree_length[control_tree_length$num== sum(i),]
  
  e=lapply(c('Prot', 'RNA','Ribo'), function(j){
  c= e[e$type%in%j,"V1"]
  b = d[j]
  if(b>mean(c)){ a=pnorm(mean = mean(c),sd = sd(c),b,lower.tail = F)}
  if(b<mean(c)){ a=pnorm(mean = mean(c),sd = sd(c),b)} 
  return(c(a, b/mean(c)))
  })
  e    
})

b=lapply(b, function(i){
  i<<-i
  unlist(i)
})

b= do.call(rbind,b)


for(i in c(1,3,5)){
  b[,i]=p.adjust(b[,i], method = 'fdr')
}

b = b[b[,1]<0.01|b[,5]<0.01|b[,3]<0.01,]

```
We detected **`r nrow(b)`** characteristics influencing gene expression evolution (either in one direction or another)

```{r tree7}

require(knitr)
colnames(b)=c('P_prot','tree length charac/random (prot)','P_RNA','tree length charac/random (RNA)','P_Ribo','tree length charac/random (Ribo)')
kable(b, caption = 'Characteristic influecing Gene expression evolution')

```

```{r save-transcriptomics, echo=F}
save.image(here::here('output',paste0(chap_num,"transcriptomics.rdata")))
```