```{r load-data-DE, echo=F}
chap_cur=07
#load(here::here('output',sprintf("%02d-differential-expression.rdata",chap_cur-1)))
```
# Functional analysis

## Functional map for differentially expressed genes

```{r treemap-bioprocess, fig.path='plot/', warning=F, out.width='100%'}
library(treemap)
library(d3Tree)
treemap(df_dfe_annot, index=c("BIOPROCESS_all", "comparison"), vSize='regulated', vColor="EffectSize", type="value") 
treemap(df_dfe_annot, index=c("BIOPROCESS_all", "GENENAME"), vSize="regulated", vColor="EffectSize", type="value")
treemap(df_dfe_annot, index=c("BIOPROCESS_all", "ORF"), vSize="regulated", vColor="EffectSize", type="value")

```

## GO enrichment

```{r go-enrich, fig.path='plot/', warning=F, out.width='100%', fig.height=10}
library(enrichplot)
library(clusterProfiler)
library(org.Sc.sgd.db)

all_genes = int_scaled_strains %>%
  as.data.frame() %>%
  rownames_to_column('ID') %>% 
  left_join(sc_identifiers,by=c('ID'='UNIPROT')) %>%
  dplyr::filter(!duplicated(GENENAME)) %>%
  mutate(GENENAME=if_na(GENENAME,ID)) %>% 
  as_tibble() %>% 
  dplyr::select(where(is.character))

FONT_SIZE=7
WRAPPING=20

genes_proteomics = all_genes$ORF
genes_dfe = dfe_lfc$ORF
genes_dfe_list = df_dfe_annot %>% group_by(ORF) %>% summarize( mean_lfc = mean(EffectSize)) %>% arrange(desc(mean_lfc)) %>% pull(mean_lfc,ORF) 

gs_cc <- gseGO(geneList= genes_dfe_list, OrgDb = org.Sc.sgd.db, keyType = "ORF", 
              ont = "CC", minGSSize = 20, maxGSSize = 500, pvalueCutoff = 0.05)
gs_mf <- gseGO(geneList= genes_dfe_list, OrgDb = org.Sc.sgd.db, keyType = "ORF", 
              ont = "MF", minGSSize = 20, maxGSSize = 500, pvalueCutoff = 0.05)
gs_bp <- gseGO(geneList= genes_dfe_list, OrgDb = org.Sc.sgd.db, keyType = "ORF", 
              ont = "BP", minGSSize = 20, maxGSSize = 500, pvalueCutoff = 0.05)

if( has_enriched(gs_mf) ){
  dotplot(gs_mf,title='GeneSet enrichment - Molecular Function',color='p.adjust',label_format=WRAPPING,font.size=FONT_SIZE)
}
if( has_enriched(gs_bp) ){
  dotplot(gs_bp,title='GeneSet enrichment - Biological Process',color='p.adjust',label_format=WRAPPING,font.size=FONT_SIZE)
}
if( has_enriched(gs_cc) ){
  dotplot(gs_cc,title='GeneSet enrichment - Cellular Component',color='p.adjust',label_format=WRAPPING,font.size=FONT_SIZE)
}
```

## KEGG pathway enrichment

```{r kegg-enrich, fig.path='plot/', warning=F, out.width='100%', fig.height=10}
genes_proteomics_uni = all_genes$ID
genes_dfe_uni = dfe_lfc$ID
genes_dfe_list_uni = df_dfe_annot %>% group_by(uniprot) %>% summarize( mean_lfc = mean(EffectSize)) %>% arrange(desc(mean_lfc)) %>% pull(mean_lfc,uniprot)

dfe_kk = clusterProfiler::enrichKEGG(genes_dfe_uni,organism = 'sce', keyType = 'uniprot', pvalueCutoff = 0.05, universe =genes_proteomics_uni)
dotplot(dfe_kk,title='KEGG enrichment',color='p.adjust',label_format=WRAPPING,font.size=FONT_SIZE)
#heatplot(dfe_kk, foldChange=genes_dfe_list_uni)

dfe_mf = clusterProfiler::enrichGO(genes_dfe,universe =genes_proteomics, 
                                OrgDb = org.Sc.sgd.db, keyType = 'ORF',
                                ont = 'MF', pool = F,pvalueCutoff = 0.05)
dfe_bp = clusterProfiler::enrichGO(genes_dfe,universe =genes_proteomics, 
                                OrgDb = org.Sc.sgd.db, keyType = 'ORF',
                                ont = 'BP', pool = F,pvalueCutoff = 0.05)
dfe_cc = clusterProfiler::enrichGO(genes_dfe,universe =genes_proteomics, 
                                OrgDb = org.Sc.sgd.db, keyType = 'ORF',
                                ont = 'CC', pool = F,pvalueCutoff = 0.05)


if( has_enriched(dfe_mf)  ){
  dotplot(dfe_mf,title='GO enrichment - Molecular Function',color='p.adjust',label_format=WRAPPING,font.size=FONT_SIZE)
}
if( has_enriched(dfe_bp) ){
  dotplot(dfe_bp,title='GO enrichment - Biological Process',color='p.adjust',label_format=WRAPPING,font.size=FONT_SIZE)
}
if( has_enriched(dfe_cc) ){
  dotplot(dfe_cc,title='GO enrichment - Cellular Component',color='p.adjust',label_format=WRAPPING,font.size=FONT_SIZE,)
}
```

## GO enrichment for clusters of differentially expressed genes

```{r func-enrich-clust, fig.path='plot/', warning=F, out.width='100%', fig.height=10}
dfe_lfc$cluster = as.factor(cl_prot)
dfe_cluster = dfe_lfc %>% drop_na(ORF)
genes_dfe_cluster = split(dfe_cluster$ORF,dfe_cluster$cluster)
genes_dfe_cluster_uni = split(dfe_lfc$ID,dfe_lfc$cluster)

# cl_go <- compareCluster(geneCluster = genes_dfe_cluster, fun = enrichGO, universe=genes_proteomics,
#                      OrgDb= org.Sc.sgd.db, keyType= "ORF", ont='ALL', pool=F,
#                      pAdjustMethod = "BH", pvalueCutoff  = 0.05)
# dotplot(cl_go,split='ONTOLOGY',color='p.adjust',label_format=WRAPPING,font.size=FONT_SIZE)
# 

cl_cc <- compareCluster(geneCluster = genes_dfe_cluster, fun = enrichGO, universe=genes_proteomics,
                     OrgDb= org.Sc.sgd.db, keyType= "ORF", ont='CC', pool=F,
                     pAdjustMethod = "BH", pvalueCutoff  = 0.05)
cl_bp <- compareCluster(geneCluster = genes_dfe_cluster, fun = enrichGO, universe=genes_proteomics,
                     OrgDb= org.Sc.sgd.db, keyType= "ORF", ont='BP', pool=F,
                     pAdjustMethod = "BH", pvalueCutoff  = 0.05)
cl_mf <- compareCluster(geneCluster = genes_dfe_cluster, fun = enrichGO, universe=genes_proteomics,
                     OrgDb= org.Sc.sgd.db, keyType= "ORF", ont='MF', pool=F,
                     pAdjustMethod = "BH", pvalueCutoff  = 0.05)

if( has_enriched(cl_mf)  ){
dotplot(cl_mf,title='Cluster enrichment - Molecular Function',color='p.adjust',label_format=WRAPPING,font.size=FONT_SIZE)
}
if( has_enriched(cl_bp) ){
  dotplot(cl_bp,title='Cluster enrichment - Biological Process',color='p.adjust',label_format=WRAPPING,font.size=FONT_SIZE)
}
if( has_enriched(cl_cc) ){
  dotplot(cl_cc,title='Cluster enrichment - Cellular Component',color='p.adjust',label_format=WRAPPING,font.size=FONT_SIZE)
}
```


```{r save-data-func-analysis, echo=F}
save.image(here('output',sprintf("%02d-func-analysis.rdata",chap_cur)))
```