<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 2 Proteomics experiment | Intraspecies proteome expression</title>
<meta name="author" content="Benjamin Dubreuil">
<meta name="description" content="2.1 Experiment design For this experiment, we selected eight strains from a population of 1,011 S. cerevisiae isolates, representing the overall ecological, geographical and genetic diversity...">
<meta name="generator" content="bookdown 0.26 with bs4_book()">
<meta property="og:title" content="Chapter 2 Proteomics experiment | Intraspecies proteome expression">
<meta property="og:type" content="book">
<meta property="og:description" content="2.1 Experiment design For this experiment, we selected eight strains from a population of 1,011 S. cerevisiae isolates, representing the overall ecological, geographical and genetic diversity...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 2 Proteomics experiment | Intraspecies proteome expression">
<meta name="twitter:description" content="2.1 Experiment design For this experiment, we selected eight strains from a population of 1,011 S. cerevisiae isolates, representing the overall ecological, geographical and genetic diversity...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.22/datatables.js"></script><link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="libs/dt-ext-fixedheader-1.11.3/css/fixedHeader.dataTables.min.css" rel="stylesheet">
<script src="libs/dt-ext-fixedheader-1.11.3/js/dataTables.fixedHeader.min.js"></script><link href="libs/dt-ext-fixedcolumns-1.11.3/css/fixedColumns.dataTables.min.css" rel="stylesheet">
<script src="libs/dt-ext-fixedcolumns-1.11.3/js/dataTables.fixedColumns.min.js"></script><script src="libs/jszip-1.11.3/jszip.min.js"></script><link href="libs/dt-ext-buttons-1.11.3/css/buttons.dataTables.min.css" rel="stylesheet">
<script src="libs/dt-ext-buttons-1.11.3/js/dataTables.buttons.min.js"></script><script src="libs/dt-ext-buttons-1.11.3/js/buttons.html5.min.js"></script><script src="libs/dt-ext-buttons-1.11.3/js/buttons.colVis.min.js"></script><script src="libs/dt-ext-buttons-1.11.3/js/buttons.print.min.js"></script><link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet">
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script><link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet">
<script src="libs/selectize-0.12.0/selectize.min.js"></script><link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Intraspecies proteome expression</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Introduction</a></li>
<li><a class="active" href="proteomics-experiment.html"><span class="header-section-number">2</span> Proteomics experiment</a></li>
<li><a class="" href="data-processing.html"><span class="header-section-number">3</span> Data processing</a></li>
<li><a class="" href="qc.html"><span class="header-section-number">4</span> Quality control</a></li>
<li><a class="" href="strains-expression-comparison.html"><span class="header-section-number">5</span> Strains expression comparison</a></li>
<li><a class="" href="differential-expression.html"><span class="header-section-number">6</span> Differential Expression</a></li>
<li><a class="" href="session-info.html"><span class="header-section-number">7</span> Session Info</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/benjamin-elusers/intraspecies-proteomics">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="proteomics-experiment" class="section level1" number="2">
<h1>
<span class="header-section-number">2</span> Proteomics experiment<a class="anchor" aria-label="anchor" href="#proteomics-experiment"><i class="fas fa-link"></i></a>
</h1>
<div id="experiment-design" class="section level2" number="2.1">
<h2>
<span class="header-section-number">2.1</span> Experiment design<a class="anchor" aria-label="anchor" href="#experiment-design"><i class="fas fa-link"></i></a>
</h2>
<p>For this experiment, we selected eight strains from a population
of 1,011 <em>S. cerevisiae</em> isolates, representing the overall ecological, geographical
and genetic diversity (Figure <a href="proteomics-experiment.html#fig:phylo-8-strain">2.1</a>.</p>
<p>The eight strains were grown on synthetic defined media (SD). We carefully monitored
the strains growth to harvest cells close to the mid-log phase (OD ~ 0.5). Then,
we proceed to wash the samples in PBS and flash-freeze cell pellets for a
whole-lysate proteomics profiling.</p>
<p>For each strains, we prepared biological replicates that originated from two
distinct colonies which and two technical replicates.</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:phylo-8-strain"></span>
<img src="images/phylogenetic-tree-8-strains.png" alt="Phylogenetic tree of 1011 *cerevisiae* isolates highlighting the 8 strains used for RNASeq/RiboSeq/proteomics exepriment" width="40%"><p class="caption">
Figure 2.1: Phylogenetic tree of 1011 <em>cerevisiae</em> isolates highlighting the 8 strains used for RNASeq/RiboSeq/proteomics exepriment
</p>
</div>
<p>Altogether there were 32 samples of yeast cell pellets including 4 replicates
(two biological and two technical) that we sent to the Weizmann proteomics unit.
(<a href="https://g-incpm.weizmann.ac.il/units/deBottonProteinProfiling/how-does-it-work%3F">INPCM</a>).</p>
<p>The samples submitted correspond to 4mL cultures with OD ranging from 0.4-0.8.</p>
<p><strong>OD of samples (table)</strong></p>
</div>
<div id="sample-preparation" class="section level2" number="2.2">
<h2>
<span class="header-section-number">2.2</span> Sample preparation<a class="anchor" aria-label="anchor" href="#sample-preparation"><i class="fas fa-link"></i></a>
</h2>
<p>The cell pellets were subjected to lysis and in solution tryptic digestion using
the S-Trap method (by Protifi) followed by a solid phase extraction cleaning step (Oasis HLB).</p>
</div>
<div id="liquid-chromatography-mass-spectrometry" class="section level2" number="2.3">
<h2>
<span class="header-section-number">2.3</span> Liquid chromatography mass spectrometry<a class="anchor" aria-label="anchor" href="#liquid-chromatography-mass-spectrometry"><i class="fas fa-link"></i></a>
</h2>
<p>The resulting peptides were analyzed using nanoflow liquid chromatography
(nanoAcquity) coupled to high resolution, high mass accuracy mass spectrometry
(Thermo Exploris 480). Each sample was analyzed on the instrument separately in
a random order in discovery mode.</p>
</div>
<div id="peptide-identification-and-quantification" class="section level2" number="2.4">
<h2>
<span class="header-section-number">2.4</span> Peptide identification and quantification<a class="anchor" aria-label="anchor" href="#peptide-identification-and-quantification"><i class="fas fa-link"></i></a>
</h2>
<p>Raw data was processed with MaxQuant v1.6.6.0. The data were searched with the
Andromeda search engine against a database containing protein sequences of
<em>Saccharomyces cerevisiae</em> as downloaded from Uniprot.org, and appended with
common lab protein contaminants.</p>
<p>The following modifications were defined for the search:
Fixed modification- cysteine carbamidomethylation.
Variable modifications- methionine oxidation and protein N-terminal acetylation.</p>
<p>The quantitative comparisons were calculated using Perseus v1.6.0.7.
Decoy hits were filtered out and only proteins that were detected
in at least two replicates of at least one experimental group were kept.</p>
</div>
<div id="loading-mass-spectrometry-results" class="section level2" number="2.5">
<h2>
<span class="header-section-number">2.5</span> Loading Mass Spectrometry Results<a class="anchor" aria-label="anchor" href="#loading-mass-spectrometry-results"><i class="fas fa-link"></i></a>
</h2>
<p>First, we read the MaxQuant output file (CSV/XLSX format) containing all the hits.
A hit corresponds to the set of peptides that were matched to a single protein group.
<em>Note: protein groups may contain one or multiple identified proteins (including contaminants)</em></p>
<pre><code>#&gt; Total number of proteins hits: 3635
#&gt; Sample names:
#&gt;    1. AMH_1_1_12
#&gt;   2. AMH_1_2_10
#&gt;   3. AMH_2_1_12
#&gt;   4. AMH_2_2_12
#&gt;   5. BAN_1_1_12
#&gt;   6. BAN_1_2_10
#&gt;   7. BAN_2_1_10
#&gt;   8. BAN_2_2_10
#&gt;   9. BED_1_1_12
#&gt;  10. BED_1_2_10
#&gt;  11. BED_2_1_12
#&gt;  12. BED_2_2_10
#&gt;  13. BPL_1_1_10
#&gt;  14. BPL_1_2_12
#&gt;  15. BPL_2_1_10
#&gt;  16. BPL_2_2_12
#&gt;  17. BTT_1_1_12
#&gt;  18. BTT_1_2_10
#&gt;  19. BTT_2_1_12
#&gt;  20. BTT_2_2_12
#&gt;  21. CMP_1_1_10
#&gt;  22. CMP_1_2_12
#&gt;  23. CMP_2_1_10
#&gt;  24. CMP_2_2_10
#&gt;  25. CPI_1_1_12
#&gt;  26. CPI_1_2_10
#&gt;  27. CPI_2_1_12
#&gt;  28. CPI_2_2_12
#&gt;  29. CQC_1_1_10
#&gt;  30. CQC_1_2_12
#&gt;  31. CQC_2_1_10
#&gt;  32. CQC_2_2_10</code></pre>
<p>Statistics of proteomics hits:</p>
<p><strong>3635</strong> total hits</p>
<ul>
<li>
<strong>20</strong> contaminants</li>
<li>
<strong>53</strong> reversed sequences</li>
<li>
<strong>3562</strong> proteins identified:
<ul>
<li>
<strong>3485</strong> unique proteins</li>
<li>
<strong>77</strong> multiple proteins (with 77 duplicated pairs)</li>
</ul>
</li>
</ul>
<p><strong>In this proteomics experiment, a large number of proteins were identified and quantified.
The quantified cellular proteome roughly amounts to half of the reference proteome sequences
of the yeast <em>S. cerevisiae</em> S288C</strong></p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="index.html"><span class="header-section-number">1</span> Introduction</a></div>
<div class="next"><a href="data-processing.html"><span class="header-section-number">3</span> Data processing</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#proteomics-experiment"><span class="header-section-number">2</span> Proteomics experiment</a></li>
<li><a class="nav-link" href="#experiment-design"><span class="header-section-number">2.1</span> Experiment design</a></li>
<li><a class="nav-link" href="#sample-preparation"><span class="header-section-number">2.2</span> Sample preparation</a></li>
<li><a class="nav-link" href="#liquid-chromatography-mass-spectrometry"><span class="header-section-number">2.3</span> Liquid chromatography mass spectrometry</a></li>
<li><a class="nav-link" href="#peptide-identification-and-quantification"><span class="header-section-number">2.4</span> Peptide identification and quantification</a></li>
<li><a class="nav-link" href="#loading-mass-spectrometry-results"><span class="header-section-number">2.5</span> Loading Mass Spectrometry Results</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/benjamin-elusers/intraspecies-proteomics/blob/master/01-proteomics.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/benjamin-elusers/intraspecies-proteomics/edit/master/01-proteomics.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Intraspecies proteome expression</strong>" was written by Benjamin Dubreuil. It was last built on 19-May-2022 18:15.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
