<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 7 Quality control | Intraspecies proteome expression</title>
<meta name="author" content="Benjamin Dubreuil">
<meta name="author" content="Elie Teyssonniere">
<meta name="description" content="First, we retrieve the LFQ-intensities of each protein hit (label-free quantitation). # get lfq-peptide intensities (lfq=label-free quantitation) intensities = ms1 %&gt;%...">
<meta name="generator" content="bookdown 0.26 with bs4_book()">
<meta property="og:title" content="Chapter 7 Quality control | Intraspecies proteome expression">
<meta property="og:type" content="book">
<meta property="og:description" content="First, we retrieve the LFQ-intensities of each protein hit (label-free quantitation). # get lfq-peptide intensities (lfq=label-free quantitation) intensities = ms1 %&gt;%...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 7 Quality control | Intraspecies proteome expression">
<meta name="twitter:description" content="First, we retrieve the LFQ-intensities of each protein hit (label-free quantitation). # get lfq-peptide intensities (lfq=label-free quantitation) intensities = ms1 %&gt;%...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.22/datatables.js"></script><link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet">
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script><link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet">
<script src="libs/selectize-0.12.0/selectize.min.js"></script><link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="libs/plotly-binding-4.10.0/plotly.js"></script><script src="libs/typedarray-0.1/typedarray.min.js"></script><link href="libs/plotly-htmlwidgets-css-2.5.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="libs/plotly-main-2.5.1/plotly-latest.min.js"></script><link href="libs/dt-ext-fixedheader-1.11.3/css/fixedHeader.dataTables.min.css" rel="stylesheet">
<script src="libs/dt-ext-fixedheader-1.11.3/js/dataTables.fixedHeader.min.js"></script><link href="libs/dt-ext-fixedcolumns-1.11.3/css/fixedColumns.dataTables.min.css" rel="stylesheet">
<script src="libs/dt-ext-fixedcolumns-1.11.3/js/dataTables.fixedColumns.min.js"></script><script src="libs/jszip-1.11.3/jszip.min.js"></script><link href="libs/dt-ext-buttons-1.11.3/css/buttons.dataTables.min.css" rel="stylesheet">
<script src="libs/dt-ext-buttons-1.11.3/js/dataTables.buttons.min.js"></script><script src="libs/dt-ext-buttons-1.11.3/js/buttons.html5.min.js"></script><script src="libs/dt-ext-buttons-1.11.3/js/buttons.colVis.min.js"></script><script src="libs/dt-ext-buttons-1.11.3/js/buttons.print.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Intraspecies proteome expression</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Introduction</a></li>
<li><a class="" href="transcriptomics.html"><span class="header-section-number">2</span> Transcriptomics</a></li>
<li><a class="" href="post-transcriptional-buffering.html"><span class="header-section-number">3</span> Post transcriptional buffering</a></li>
<li><a class="" href="expression-evolution-on-the-different-layers.html"><span class="header-section-number">4</span> Expression evolution on the different layers</a></li>
<li><a class="" href="proteomics-experiment.html"><span class="header-section-number">5</span> Proteomics experiment</a></li>
<li><a class="" href="data-processing.html"><span class="header-section-number">6</span> Data processing</a></li>
<li><a class="active" href="qc.html"><span class="header-section-number">7</span> Quality control</a></li>
<li><a class="" href="strains-expression-comparison.html"><span class="header-section-number">8</span> Strains expression comparison</a></li>
<li><a class="" href="differential-expression.html"><span class="header-section-number">9</span> Differential Expression</a></li>
<li><a class="" href="functional-analysis.html"><span class="header-section-number">10</span> Functional analysis</a></li>
<li><a class="" href="session-info.html"><span class="header-section-number">11</span> Session Info</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/benjamin-elusers/intraspecies-proteomics">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="qc" class="section level1" number="7">
<h1>
<span class="header-section-number">7</span> Quality control<a class="anchor" aria-label="anchor" href="#qc"><i class="fas fa-link"></i></a>
</h1>
<p>First, we retrieve the LFQ-intensities of each protein hit (label-free quantitation).</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># get lfq-peptide intensities (lfq=label-free quantitation) </span>
<span class="va">intensities</span> <span class="op">=</span> <span class="va">ms1</span> <span class="op"><a href="https://rdrr.io/pkg/DT/man/DT-imports.html">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span>uniprot<span class="op">=</span><span class="va">majority_protein_i_ds</span>,<span class="fu">starts_with</span><span class="op">(</span><span class="st">"lfq"</span><span class="op">)</span><span class="op">)</span> 
<span class="co"># Convert intensities to long format </span>
<span class="va">long_int_all</span> <span class="op">=</span> <span class="fu">get_long_intensities</span><span class="op">(</span><span class="va">intensities</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/DT/man/DT-imports.html">%&gt;%</a></span> <span class="fu">mutate</span><span class="op">(</span>int2use <span class="op">=</span> <span class="va">log10_int</span><span class="op">)</span></code></pre></div>
<div id="total-sample-intensities" class="section level2" number="7.1">
<h2>
<span class="header-section-number">7.1</span> Total sample intensities<a class="anchor" aria-label="anchor" href="#total-sample-intensities"><i class="fas fa-link"></i></a>
</h2>
<pre><code>#&gt; Joining, by = "sample"</code></pre>
<div class="figure">
<img src="Intraspecies-proteome-expression_files/figure-html/tot_int-1.png" alt="Total intensities per sample" width="100%"><p class="caption">
(#fig:tot_int)Total intensities per sample
</p>
</div>
</div>
<div id="boxplot-of-sample-raw-intensities" class="section level2" number="7.2">
<h2>
<span class="header-section-number">7.2</span> Boxplot of sample raw intensities<a class="anchor" aria-label="anchor" href="#boxplot-of-sample-raw-intensities"><i class="fas fa-link"></i></a>
</h2>
<p>We want to inspect the distribution of peptide intensities between strains.
In addition, we will also observe in how many replicates each hit was quantified.</p>
<p><em>The distribution of peptide intensities between all strains does not show strong differences of expression:</em></p>
<div class="figure">
<img src="Intraspecies-proteome-expression_files/figure-html/boxplot-int_ub-1.png" alt="Distribution of expression for ubiquitous hits (i.e. detected in all strains)" width="100%"><p class="caption">
(#fig:boxplot-int_ub)Distribution of expression for ubiquitous hits (i.e.Â detected in all strains)
</p>
</div>
The median peptide intensity are remarkably similar (~10<sup>8</sup> or 7.9 in log10) between strains:
<div class="inline-table"><table class=" lightable-paper lightable-striped table" style='font-family: "Arial Narrow", arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; margin-left: auto; margin-right: auto;'>
<thead><tr>
<th style="text-align:left;">
strain name
</th>
<th style="text-align:right;">
AMH
</th>
<th style="text-align:right;">
BAN
</th>
<th style="text-align:right;">
BED
</th>
<th style="text-align:right;">
BPL
</th>
<th style="text-align:right;">
BTT
</th>
<th style="text-align:right;">
CMP
</th>
<th style="text-align:right;">
CPI
</th>
<th style="text-align:right;">
CQC
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;">
median_exp
</td>
<td style="text-align:right;">
7.9
</td>
<td style="text-align:right;">
7.92
</td>
<td style="text-align:right;">
7.92
</td>
<td style="text-align:right;">
7.9
</td>
<td style="text-align:right;">
7.92
</td>
<td style="text-align:right;">
7.95
</td>
<td style="text-align:right;">
7.91
</td>
<td style="text-align:right;">
7.97
</td>
</tr></tbody>
</table></div>
<em>However, the peptide intensities can vary widely (&gt;10-fold) when a hit is not detected across all replicates:</em>
<div class="inline-table"><table class=" lightable-paper lightable-striped table" style='font-family: "Arial Narrow", arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; margin-left: auto; margin-right: auto;'>
<thead><tr>
<th style="text-align:left;">
# replicates
</th>
<th style="text-align:right;">
1
</th>
<th style="text-align:right;">
2
</th>
<th style="text-align:right;">
3
</th>
<th style="text-align:right;">
4
</th>
</tr></thead>
<tbody><tr>
<td style="text-align:left;">
median_exp
</td>
<td style="text-align:right;">
6.82
</td>
<td style="text-align:right;">
6.92
</td>
<td style="text-align:right;">
7.04
</td>
<td style="text-align:right;">
8
</td>
</tr></tbody>
</table></div>
<p><strong>On average, peptide intensities are higher when a hit is found in more than one replicate</strong></p>
</div>
<div id="count-of-missing-hits-per-sample" class="section level2" number="7.3">
<h2>
<span class="header-section-number">7.3</span> Count of missing hits per sample<a class="anchor" aria-label="anchor" href="#count-of-missing-hits-per-sample"><i class="fas fa-link"></i></a>
</h2>
<p>The following barplot shows the range of missing hits per samples.</p>
<pre><code>#&gt; Joining, by = "sample"</code></pre>
<div class="figure">
<span style="display:block;" id="fig:missing-hit-samples"></span>
<img src="Intraspecies-proteome-expression_files/figure-html/missing-hit-samples-1.png" alt="Count of missing hits per samples" width="100%"><p class="caption">
Figure 7.1: Count of missing hits per samples
</p>
</div>
<p>CQC and CMP are the two strains containing the most missing values.</p>
<p>Particularly, the first biological replicate of those strains have twice more missing
protein intensities (509 hits so 10-16% NAs) while the rest of the samples
have on average about only 231 hits missing (4% to 9% NAs).</p>
<p>We then look at hits quantified as a function of the number of strains in which
at least one peptide was quantified. (non-ubiquitous hits)</p>
<div class="figure">
<span style="display:block;" id="fig:missing-hit"></span>
<img src="Intraspecies-proteome-expression_files/figure-html/missing-hit-1.png" alt="Distribution of expression for hits not detected in all strains" width="100%"><p class="caption">
Figure 7.2: Distribution of expression for hits not detected in all strains
</p>
</div>
<div class="inline-table"><table class=" lightable-paper lightable-striped table" style='font-family: "Arial Narrow", arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; margin-left: auto; margin-right: auto;'>
<thead><tr>
<th style="text-align:left;">
# proteins
</th>
<th style="text-align:left;">
# expressing strains
</th>
<th style="text-align:left;">
median exp. (log10)
</th>
<th style="text-align:left;">
% quantified
</th>
<th style="text-align:left;">
cum. %
</th>
<th style="text-align:left;">
cum. % (decreasing)
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
2280
</td>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
8.20
</td>
<td style="text-align:left;">
66.5%
</td>
<td style="text-align:left;">
100.0%
</td>
<td style="text-align:left;">
66.5%
</td>
</tr>
<tr>
<td style="text-align:left;">
305
</td>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
7.45
</td>
<td style="text-align:left;">
8.9%
</td>
<td style="text-align:left;">
33.5%
</td>
<td style="text-align:left;">
75.4%
</td>
</tr>
<tr>
<td style="text-align:left;">
209
</td>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
7.24
</td>
<td style="text-align:left;">
6.1%
</td>
<td style="text-align:left;">
24.6%
</td>
<td style="text-align:left;">
81.5%
</td>
</tr>
<tr>
<td style="text-align:left;">
144
</td>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
7.12
</td>
<td style="text-align:left;">
4.2%
</td>
<td style="text-align:left;">
18.5%
</td>
<td style="text-align:left;">
85.7%
</td>
</tr>
<tr>
<td style="text-align:left;">
111
</td>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
7.02
</td>
<td style="text-align:left;">
3.2%
</td>
<td style="text-align:left;">
14.3%
</td>
<td style="text-align:left;">
88.9%
</td>
</tr>
<tr>
<td style="text-align:left;">
96
</td>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
6.99
</td>
<td style="text-align:left;">
2.8%
</td>
<td style="text-align:left;">
11.1%
</td>
<td style="text-align:left;">
91.7%
</td>
</tr>
<tr>
<td style="text-align:left;">
90
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
6.96
</td>
<td style="text-align:left;">
2.6%
</td>
<td style="text-align:left;">
8.3%
</td>
<td style="text-align:left;">
94.3%
</td>
</tr>
<tr>
<td style="text-align:left;">
84
</td>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
6.89
</td>
<td style="text-align:left;">
2.4%
</td>
<td style="text-align:left;">
5.7%
</td>
<td style="text-align:left;">
96.8%
</td>
</tr>
<tr>
<td style="text-align:left;">
110
</td>
<td style="text-align:left;">
0
</td>
<td style="text-align:left;">
6.78
</td>
<td style="text-align:left;">
3.2%
</td>
<td style="text-align:left;">
3.2%
</td>
<td style="text-align:left;">
100.0%
</td>
</tr>
</tbody>
</table></div>
<p><strong>Proteins expressed across less strains have lower median peptide intensity.</strong>
Nevertheless, almost two thirds (<em>66.5%</em>) of the protein hits were detected in all strains.
About 86% protein hits were expressed in the majority of the strains (at least 5 out of 8 strains).</p>
</div>
<div id="missing-hits-across-samples" class="section level2" number="7.4">
<h2>
<span class="header-section-number">7.4</span> Missing hits across samples<a class="anchor" aria-label="anchor" href="#missing-hits-across-samples"><i class="fas fa-link"></i></a>
</h2>
<p>Furthermore, we can check whether the proportion of missing values is equally
distributed among all samples.</p>
<div class="figure">
<span style="display:block;" id="fig:na-samples-1"></span>
<img src="Intraspecies-proteome-expression_files/figure-html/na-samples-1.png" alt="Number of quantified intensities across samples" width="100%"><p class="caption">
Figure 7.3: Number of quantified intensities across samples
</p>
</div>
<pre><code>#&gt; Saving 2400 x 7200 px image</code></pre>
<div class="figure">
<span style="display:block;" id="fig:na-samples-2"></span>
<img src="plot/upset-samples-combination-nacount.pdf" alt="Number of quantified intensities across samples" width="100%"><p class="caption">
Figure 7.4: Number of quantified intensities across samples
</p>
</div>
</div>
<div id="coefficient-of-variations" class="section level2" number="7.5">
<h2>
<span class="header-section-number">7.5</span> Coefficient of variations<a class="anchor" aria-label="anchor" href="#coefficient-of-variations"><i class="fas fa-link"></i></a>
</h2>
<p>Coefficient of variations correspond to the percent of variance relative to the mean.
The following boxplots show how variable protein expression is, across strains
or across biological replicates, before and after normalization.</p>
<div class="figure">
<span style="display:block;" id="fig:boxplot-cv"></span>
<img src="Intraspecies-proteome-expression_files/figure-html/boxplot-cv-1.png" alt="Coefficient of variations (%) across samples for each normalization method" width="100%"><p class="caption">
Figure 7.5: Coefficient of variations (%) across samples for each normalization method
</p>
</div>
</div>
<div id="expression-distributions" class="section level2" number="7.6">
<h2>
<span class="header-section-number">7.6</span> Expression distributions<a class="anchor" aria-label="anchor" href="#expression-distributions"><i class="fas fa-link"></i></a>
</h2>
<p>To highlight the strength of normalization, we also show the density distribution of
expression before and after normalization using each of the following normalization methods:</p>
<div class="inline-table"><table class=" lightable-paper lightable-striped table" style='font-family: "Arial Narrow", arial, helvetica, sans-serif; width: auto !important; margin-left: auto; margin-right: auto; '>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:left;">
Normalizations:
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
1
</td>
<td style="text-align:left;">
log2
</td>
</tr>
<tr>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
VSN
</td>
</tr>
<tr>
<td style="text-align:left;">
3
</td>
<td style="text-align:left;">
GI
</td>
</tr>
<tr>
<td style="text-align:left;">
4
</td>
<td style="text-align:left;">
median
</td>
</tr>
<tr>
<td style="text-align:left;">
5
</td>
<td style="text-align:left;">
mean
</td>
</tr>
<tr>
<td style="text-align:left;">
6
</td>
<td style="text-align:left;">
Quantile
</td>
</tr>
<tr>
<td style="text-align:left;">
7
</td>
<td style="text-align:left;">
CycLoess
</td>
</tr>
<tr>
<td style="text-align:left;">
8
</td>
<td style="text-align:left;">
RLR
</td>
</tr>
</tbody>
</table></div>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">draw_normalization_density</span><span class="op">(</span><span class="va">int_raw</span>,<span class="va">int_lfq</span>,<span class="va">df.group</span><span class="op">)</span>
<span class="co">#&gt; Input data checked. All fields are valid.</span>
<span class="co">#&gt; Sample check: More than one sample group found</span>
<span class="co">#&gt; Sample replication check: All samples have replicates</span>
<span class="co">#&gt; No RT column found, skipping RT processing</span>
<span class="co">#&gt; No RT column specified (column named 'RT') or option not specified Skipping RT normalization.</span>
<span class="co">#&gt; Joining, by = "sample"</span>
<span class="co">#&gt; Joining, by = "sample"</span></code></pre></div>
<div class="inline-figure"><img src="Intraspecies-proteome-expression_files/figure-html/exp_density-1.png" width="100%"></div>
</div>
<div id="compare-all-vs-all-expression" class="section level2" number="7.7">
<h2>
<span class="header-section-number">7.7</span> Compare all-vs-all expression<a class="anchor" aria-label="anchor" href="#compare-all-vs-all-expression"><i class="fas fa-link"></i></a>
</h2>
<div id="scatterplots" class="section level3" number="7.7.1">
<h3>
<span class="header-section-number">7.7.1</span> Scatterplots<a class="anchor" aria-label="anchor" href="#scatterplots"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># scatterplots between all samples</span>
<span class="va">scmat_all</span><span class="op">=</span><span class="fu">draw_scatterplots</span><span class="op">(</span>datain<span class="op">=</span><span class="va">int_all</span><span class="op">)</span>
<span class="co">#print(scmat_all)</span>

<span class="fu">ggsave</span><span class="op">(</span><span class="va">scmat_all</span>,filename<span class="op">=</span><span class="fu">here</span><span class="fu">::</span><span class="fu"><a href="https://here.r-lib.org//reference/here.html">here</a></span><span class="op">(</span><span class="st">'plot'</span>,<span class="st">'scatterplot-samples-intensities.pdf'</span><span class="op">)</span>,width <span class="op">=</span> <span class="fl">1200</span>, units<span class="op">=</span><span class="st">'px'</span><span class="op">)</span>
<span class="co">#&gt; Saving 1200 x 4500 px image</span>

<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/include_graphics.html">include_graphics</a></span><span class="op">(</span><span class="fu">here</span><span class="op">(</span><span class="st">'plot'</span>,<span class="st">'scatterplot-samples-intensities.pdf'</span><span class="op">)</span>,auto_pdf <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<embed src="/home/benjamin/Desktop/GitHub/intraspecies-proteomics/plot/scatterplot-samples-intensities.pdf" width="100%" type="application/pdf"></embed>
</div>
<div id="heatmap-correlation" class="section level3" number="7.7.2">
<h3>
<span class="header-section-number">7.7.2</span> Heatmap correlation<a class="anchor" aria-label="anchor" href="#heatmap-correlation"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># heatmap correlation</span>
<span class="va">cs_all</span><span class="op">=</span><span class="fu">compute_samples_correlation</span><span class="op">(</span><span class="va">int_all</span><span class="op">)</span>
<span class="co">#&gt; Compute pairwise samples correlation (Spearman)...</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Correlation method: 'spearman'</span>
<span class="co">#&gt; Missing treated using: 'pairwise.complete'</span>
<span class="va">COR_RANGE</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span><span class="op">(</span> <span class="va">cs_all</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/row.html">row</a></span><span class="op">(</span><span class="va">cs_all</span><span class="op">)</span> <span class="op">==</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/col.html">col</a></span><span class="op">(</span><span class="va">cs_all</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span><span class="op">]</span> <span class="op">)</span>

<span class="va">by_sample</span> <span class="op">=</span> <span class="va">df.group</span> <span class="op"><a href="https://rdrr.io/pkg/DT/man/DT-imports.html">%&gt;%</a></span> <span class="fu">column_to_rownames</span><span class="op">(</span><span class="st">"sample"</span><span class="op">)</span>
<span class="va">hm_all</span><span class="op">=</span><span class="fu">draw_heatmap_samples</span><span class="op">(</span>mcor <span class="op">=</span> <span class="va">cs_all</span>,df.group<span class="op">=</span><span class="va">by_sample</span>,col.group <span class="op">=</span> <span class="va">col.group</span><span class="op">)</span></code></pre></div>
<div class="inline-figure"><img src="plot/heatmap-all-1.png" width="100%"></div>
<p>The heatmap correlations of all samples show the high correlation of expression
between replicates and between most strains:
[0.675 - 0.940]</p>
<p>The expression from the 1st biological replicates of strains CQC and CMP seem
slighlty less correlated to the other samples.</p>
</div>
<div id="principle-component-analysis" class="section level3" number="7.7.3">
<h3>
<span class="header-section-number">7.7.3</span> Principle Component Analysis<a class="anchor" aria-label="anchor" href="#principle-component-analysis"><i class="fas fa-link"></i></a>
</h3>
<p>Finally, the PCA reveals the distance between each sample.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">int_all_scaled</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/scale.html">scale</a></span><span class="op">(</span><span class="va">int_all</span>,center<span class="op">=</span><span class="cn">T</span>, scale <span class="op">=</span> <span class="cn">T</span><span class="op">)</span>
<span class="fu">make_pca</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html">na.omit</a></span><span class="op">(</span><span class="va">int_all_scaled</span><span class="op">)</span>, with_labels<span class="op">=</span><span class="cn">T</span>,col_by_group<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></code></pre></div>
<div class="inline-figure"><img src="plot/pca-all-1.png" width="100%"></div>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="data-processing.html"><span class="header-section-number">6</span> Data processing</a></div>
<div class="next"><a href="strains-expression-comparison.html"><span class="header-section-number">8</span> Strains expression comparison</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#qc"><span class="header-section-number">7</span> Quality control</a></li>
<li><a class="nav-link" href="#total-sample-intensities"><span class="header-section-number">7.1</span> Total sample intensities</a></li>
<li><a class="nav-link" href="#boxplot-of-sample-raw-intensities"><span class="header-section-number">7.2</span> Boxplot of sample raw intensities</a></li>
<li><a class="nav-link" href="#count-of-missing-hits-per-sample"><span class="header-section-number">7.3</span> Count of missing hits per sample</a></li>
<li><a class="nav-link" href="#missing-hits-across-samples"><span class="header-section-number">7.4</span> Missing hits across samples</a></li>
<li><a class="nav-link" href="#coefficient-of-variations"><span class="header-section-number">7.5</span> Coefficient of variations</a></li>
<li><a class="nav-link" href="#expression-distributions"><span class="header-section-number">7.6</span> Expression distributions</a></li>
<li>
<a class="nav-link" href="#compare-all-vs-all-expression"><span class="header-section-number">7.7</span> Compare all-vs-all expression</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#scatterplots"><span class="header-section-number">7.7.1</span> Scatterplots</a></li>
<li><a class="nav-link" href="#heatmap-correlation"><span class="header-section-number">7.7.2</span> Heatmap correlation</a></li>
<li><a class="nav-link" href="#principle-component-analysis"><span class="header-section-number">7.7.3</span> Principle Component Analysis</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/benjamin-elusers/intraspecies-proteomics/blob/master/04-QC.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/benjamin-elusers/intraspecies-proteomics/edit/master/04-QC.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Intraspecies proteome expression</strong>" was written by Benjamin Dubreuil<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>, Elie Teyssonniere<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>. It was last built on 05-Jun-2022 22:43.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
