<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 2 Transcriptomics | Intraspecies proteome expression</title>
<meta name="author" content="Benjamin Dubreuil">
<meta name="author" content="Elie Teyssonniere">
<meta name="description" content="2.1 Summary During evolution, mutations arise in genomes, potentially affecting every gene and protein at every site over time. Degrees of sequence divergence within populations and between...">
<meta name="generator" content="bookdown 0.26 with bs4_book()">
<meta property="og:title" content="Chapter 2 Transcriptomics | Intraspecies proteome expression">
<meta property="og:type" content="book">
<meta property="og:description" content="2.1 Summary During evolution, mutations arise in genomes, potentially affecting every gene and protein at every site over time. Degrees of sequence divergence within populations and between...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Chapter 2 Transcriptomics | Intraspecies proteome expression">
<meta name="twitter:description" content="2.1 Summary During evolution, mutations arise in genomes, potentially affecting every gene and protein at every site over time. Degrees of sequence divergence within populations and between...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="libs/datatables-binding-0.22/datatables.js"></script><link href="libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet">
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script><link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet">
<script src="libs/selectize-0.12.0/selectize.min.js"></script><link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<link href="libs/dt-ext-fixedheader-1.11.3/css/fixedHeader.dataTables.min.css" rel="stylesheet">
<script src="libs/dt-ext-fixedheader-1.11.3/js/dataTables.fixedHeader.min.js"></script><link href="libs/dt-ext-fixedcolumns-1.11.3/css/fixedColumns.dataTables.min.css" rel="stylesheet">
<script src="libs/dt-ext-fixedcolumns-1.11.3/js/dataTables.fixedColumns.min.js"></script><script src="libs/jszip-1.11.3/jszip.min.js"></script><link href="libs/dt-ext-buttons-1.11.3/css/buttons.dataTables.min.css" rel="stylesheet">
<script src="libs/dt-ext-buttons-1.11.3/js/dataTables.buttons.min.js"></script><script src="libs/dt-ext-buttons-1.11.3/js/buttons.html5.min.js"></script><script src="libs/dt-ext-buttons-1.11.3/js/buttons.colVis.min.js"></script><script src="libs/dt-ext-buttons-1.11.3/js/buttons.print.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">Intraspecies proteome expression</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html"><span class="header-section-number">1</span> Introduction</a></li>
<li><a class="active" href="transcriptomics.html"><span class="header-section-number">2</span> Transcriptomics</a></li>
<li><a class="" href="proteomics-experiment.html"><span class="header-section-number">3</span> Proteomics experiment</a></li>
<li><a class="" href="data-processing.html"><span class="header-section-number">4</span> Data processing</a></li>
<li><a class="" href="qc.html"><span class="header-section-number">5</span> Quality control</a></li>
<li><a class="" href="strains-expression-comparison.html"><span class="header-section-number">6</span> Strains expression comparison</a></li>
<li><a class="" href="differential-expression.html"><span class="header-section-number">7</span> Differential Expression</a></li>
<li><a class="" href="session-info.html"><span class="header-section-number">8</span> Session Info</a></li>
<li><a class="" href="references.html">References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/benjamin-elusers/intraspecies-proteomics">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="transcriptomics" class="section level1 tabset tabset-pills" number="2">
<h1>
<span class="header-section-number">2</span> Transcriptomics<a class="anchor" aria-label="anchor" href="#transcriptomics"><i class="fas fa-link"></i></a>
</h1>
<div id="summary" class="section level2" number="2.1">
<h2>
<span class="header-section-number">2.1</span> Summary<a class="anchor" aria-label="anchor" href="#summary"><i class="fas fa-link"></i></a>
</h2>
<p>During evolution, mutations arise in genomes, potentially affecting every gene
and protein at every site over time. Degrees of sequence divergence within
populations and between species reflect the interplay between several
interconnected constraints (e.g., genomic, transcriptional, functional,
biophysical, structural, environmental…) and underlie phenotypic diversity and
organism fitness. Protein expression is the main determinant of a protein’s
sequence conservation (e.g., the more a gene is expressed, the more its sequence
is conserved). Although this expression-conservation correlation is well
established, its mechanistic origin is not well understood.</p>
<p>In this perspective, analyzing variation of gene expression across every layer
could reveal the key processes driving evolution among several yeast (<em>S. cerevisiae</em>)
isolates sharing an almost identical genetic background, except for
newly acquired mutations, notably in protein-coding genes (e.g. Single
Nucleotide Variations).</p>
<p>Gene expression is a multi-layered phenomenon in which variations occur
differently at each step of the protein synthesis and life
(transcription, translation and protein abundance). The exploration of each
of these steps is therefore essential to define the constraints affecting the
final protein abundance. Given this, we sought to understand how individuals
differ in terms of translational regulation and if the translational variation
is related to the transcriptional variation.</p>
<p>Therefore, we performed a joint exploration of RNA-seq, ribosome and proteomics
profiling on 8 natural <em>Saccharomyces cerevisiae</em> yeast isolates with
widely diverse genetic and ecologic origins. We found that the evolutionary
constraints on gene expression showed different magnitudes depending on the
expression layer. Specifically, expression variations tended to be 10% lower at
the translational level, which was related to a phenomenon called
post-transcriptional buffering. This phenomenon was more likely to affect some
specific genes such as essential genes as well as genes related to protein
complexes. Surprisingly, genes being preferentially affected by the
post-transcriptional buffering tended to be less expressed than the genes not
affected by this phenomenon. Together, these results highlight that variations
in gene expression were shaped differently depending on the expression level and
multiple factors, including certain functional and physical constraints. We have
also been collecting proteomics data for these 8 isolates, which will allow us
to assess with high confidence the balance between biophysics (in particular
abundance) versus function on protein evolution.</p>
</div>
<div id="our-datasets" class="section level2" number="2.2">
<h2>
<span class="header-section-number">2.2</span> Our datasets<a class="anchor" aria-label="anchor" href="#our-datasets"><i class="fas fa-link"></i></a>
</h2>
<p><strong><em>RNA-seq &amp; Ribo-seq of 8 natural isolates of Saccharomyces cerevisiae</em></strong></p>
<p>We performed both ribosome profiling and RNA sequencing on eight <em>S. cerevisiae</em>
natural isolates coming from very diverse ecological environments and being
genetically strongly different (<a href="https://www.nature.com/articles/s41586-018-0030-5">Peter et al.,2018</a>).
These isolates were cultivated in Synthetic Complete (SC) up to mid-log phase,
harvested and flash-frozen. The RNA-seq and Ribo-seq experiment was performed in
collaboration with the Riken institute in Japan.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Our dataset</span>
<span class="va">isolates</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'CPI'</span>,<span class="st">'CMP'</span>,<span class="st">'AMH'</span>,<span class="st">'CQC'</span>,<span class="st">'BPL'</span>,<span class="st">'BTT'</span>,<span class="st">'BED'</span>,<span class="st">'BAN'</span><span class="op">)</span>
<span class="va">RNA_seq_data</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="fu">here</span><span class="op">(</span><span class="st">"data"</span>,<span class="st">'RNA_seq.RDS'</span><span class="op">)</span><span class="op">)</span>
<span class="va">Ribo_seq_data</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="fu">here</span><span class="op">(</span><span class="st">"data"</span>,<span class="st">'Ribo_seq.RDS'</span><span class="op">)</span><span class="op">)</span>
<span class="va">n</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">RNA_seq_data</span><span class="op">)</span>
<span class="va">strains</span> <span class="op">=</span> <span class="fu">fread</span><span class="op">(</span><span class="fu">here</span><span class="op">(</span><span class="st">"data"</span>,<span class="st">'strains.csv'</span><span class="op">)</span>,data.table <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>
<span class="va">strains</span> <span class="op">=</span> <span class="va">strains</span><span class="op">[</span><span class="va">strains</span><span class="op">$</span><span class="va">`Standardized name`</span><span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span><span class="va">isolates</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">4</span>,<span class="fl">9</span>,<span class="fl">10</span>,<span class="fl">11</span>,<span class="fl">16</span><span class="op">)</span><span class="op">]</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/DT">DT</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html">datatable</a></span><span class="op">(</span><span class="va">strains</span>, rownames <span class="op">=</span> <span class="cn">FALSE</span>, caption <span class="op">=</span> <span class="cn">NULL</span>,
               filter <span class="op">=</span> <span class="st">"top"</span>, escape <span class="op">=</span> <span class="cn">FALSE</span>, style <span class="op">=</span> <span class="st">"default"</span>,
               width <span class="op">=</span> <span class="cn">NULL</span>, height <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-4d75603b09f31e332b50" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4d75603b09f31e332b50">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\" disabled=\"\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["DBVPG1878","M.9.1","LCBG-3D6","CEY624","MTF2546","EN14S01","HN15","YPS615"],["BPL","BTT","CPI","BED","CQC","AMH","BAN","CMP"],["Grape must","Cachaça","Agave spp. fermentation during rustic mezcal production","Human feces","Cocoa bean fermentation","Soil","Rotten wood","Quercus sp."],["Wine","Distillery","Distillery","Human","Fermentation","Soil","Nature","Tree"],["2","2","2","2","2","2","2","2"],["euploid","euploid","euploid","euploid","euploid","euploid","euploid","euploid"],["heterozygous","heterozygous","heterozygous","heterozygous","heterozygous","homozygous","homozygous","homozygous"],["1. Wine/European (subclade 4)","3. Brazilian bioethanol","9. Mexican agave","10. French Guiana human","12. West African cocoa","17. Taiwanese","20. CHN V","21. Ecuadorean"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Isolate name<\/th>\n      <th>Standardized name<\/th>\n      <th>Isolation<\/th>\n      <th>Ecological origins<\/th>\n      <th>Ploidy<\/th>\n      <th>Aneuploidies<\/th>\n      <th>Zygosity<\/th>\n      <th>Clades<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true}},"evals":[],"jsHooks":[]}</script><p>Our data encompassed <strong>3755</strong> genes. The data was normalized using TPM
normalization where for each gene in each isolate, we divided the raw count by
the ORF length and we applied a per million factor (total read count /
1,000,000)</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">p1</span><span class="op">=</span> <span class="fu">ggcorr</span><span class="op">(</span><span class="va">RNA_seq_data</span> ,method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"everything"</span>, <span class="st">"spearman"</span><span class="op">)</span>, label <span class="op">=</span> <span class="cn">T</span>, label_round <span class="op">=</span> <span class="fl">2</span>, midpoint <span class="op">=</span> <span class="fl">0.75</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.6</span>,<span class="fl">1</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">'RNA-seq correlation matrix'</span><span class="op">)</span>
<span class="va">p2</span><span class="op">=</span><span class="fu">ggcorr</span><span class="op">(</span><span class="va">Ribo_seq_data</span> ,method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"everything"</span>, <span class="st">"spearman"</span><span class="op">)</span>, label <span class="op">=</span> <span class="cn">T</span>, label_round <span class="op">=</span> <span class="fl">2</span>, midpoint <span class="op">=</span> <span class="fl">0.75</span>, limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.6</span>,<span class="fl">1</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">+</span>
  <span class="fu">ggtitle</span><span class="op">(</span><span class="st">'Ribo-seq correlation matrix'</span><span class="op">)</span>

<span class="fu">grid.arrange</span><span class="op">(</span><span class="va">p1</span>,<span class="va">p2</span>, ncol<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<div class="inline-figure"><img src="Intraspecies-proteome-expression_files/figure-html/corr-rna-ribo-1.png" width="100%"></div>
<p>We also calculated a Translation efficiency value which correspond, for each
gene in each isolate, to the Ribo-seq TPM value divided by the RNA-seq TPM
value. In brief, this represents how well a transcript will be used for
translation.</p>
</div>
<div id="overlap-with-proteomic" class="section level2" number="2.3">
<h2>
<span class="header-section-number">2.3</span> Overlap with proteomic<a class="anchor" aria-label="anchor" href="#overlap-with-proteomic"><i class="fas fa-link"></i></a>
</h2>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="index.html"><span class="header-section-number">1</span> Introduction</a></div>
<div class="next"><a href="proteomics-experiment.html"><span class="header-section-number">3</span> Proteomics experiment</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#transcriptomics"><span class="header-section-number">2</span> Transcriptomics</a></li>
<li><a class="nav-link" href="#summary"><span class="header-section-number">2.1</span> Summary</a></li>
<li><a class="nav-link" href="#our-datasets"><span class="header-section-number">2.2</span> Our datasets</a></li>
<li><a class="nav-link" href="#overlap-with-proteomic"><span class="header-section-number">2.3</span> Overlap with proteomic</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/benjamin-elusers/intraspecies-proteomics/blob/master/01-transcriptomics.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/benjamin-elusers/intraspecies-proteomics/edit/master/01-transcriptomics.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Intraspecies proteome expression</strong>" was written by Benjamin Dubreuil<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>, Elie Teyssonniere<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>. It was last built on 30-May-2022 16:14.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
